gno.land/pkg/sdk/vm/params.go

```diff
@@ -91,6 +104,97 @@ func (vm *VMKeeper) getSysNamesPkgParam(ctx sdk.Context) string {
 	return sysNamesPkg
 }
 
-func (vm *VMKeeper) WillSetParam(ctx sdk.Context, key string, value interface{}) {
-	// XXX validate input?
+func (vm *VMKeeper) getValsetRealmParam(ctx sdk.Context) string {
+	// Fetch the latest valset realm path
+	valsetRealm := ValsetRealmDefault
+	vm.prmk.GetString(ctx, ValsetRealmParamPath, &valsetRealm)
+
+	return valsetRealm
+}
+
+var reValsetUpdates = regexp.MustCompile(fmt.Sprintf("^%s_\\d+$", "valset_updates"))
```

Comment:
It seems the only purpose of this integer is to hold two values simultaneously in the "queue." However, we are not using the height since we receive updates directly in the message.

We should process the changes in the arriving valset_updates key if they are set. Once processed, we can set valset_updates to "" and be done. Essentially, we would remove the integer suffix and maintain a static valset_updates key.

This approach resembles an event system, but it has the advantage of being usable for genesis, updates, and unit tests in the same way. WillSetParams can serve as a tool for various testing scenarios.

We act as a proxy between the VM and the TM2 valset management by applying the changes without question.

Reply:
Dropped as part of the new param system proposed in the PR comments 🙏